### PR TITLE
ENH: Better interface for model builder from logistic regression

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR adds a class wrapper for the logistic regression builder similar to the one that exists for tree-based models, and modifies the `convert_model` function to also accept logistic regression models from sklearn, which are converted to this class instead.

Since this brings up a new type of model, I also refactored the conversion from GBT and docstrings, so that now the GBT predictor can also be instantiated directly from a GBT model object, while the conversion function handles both types.

The PR includes a .rst file for the docs and adds new classes with docstrings, but it does not include it anywhere - incorporation of that file and the docstrings will be handled in https://github.com/uxlfoundation/scikit-learn-intelex/pull/2395

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

Not applicable.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
The failure in the doc CI job is expected:
```
/home/vsts/work/1/s/doc/sources/logistic_model_builder.rst: WARNING: document isn't included in any toctree [toc.not_included]
```
It will be addressed in the PR that adds daal4py docs to sklearnex: https://github.com/uxlfoundation/scikit-learn-intelex/pull/2395

**Comment by david-cortes-intel** [comment]
> Please modify examples for `log_reg_model_builder` to reflect the last changes in interface

Are you talking about the tests for logistic regression builder that were there before the introduction of the new class?

**Comment by razdoburdin** [comment]
> > Please modify examples for `log_reg_model_builder` to reflect the last changes in interface
> 
> Are you talking about the tests for logistic regression builder that were there before the introduction of the new class?

I mean this one: https://github.com/uxlfoundation/scikit-learn-intelex/blob/8b07df7a9ce46ca1d86adf1e65dfb19fabc6892a/examples/mb/log_reg_model_builder.py#L33

**Comment by david-cortes-intel** [comment]
Updated the example.

**Review by razdoburdin — APPROVED** [review]
nice!
